packages/dear-imgui/cpp/include/reactimgui.h

```diff
@@ -18,11 +18,26 @@ class StyledWidget;
 class LayoutNode;
 struct BaseStyle;
 
+enum ElementOp {
+    OpCreateElement,
+    OpPatchElement,
+    OpSetChildren,
+    OpAppendChild
+};
+
+struct ElementOpDef {
+    ElementOp op;
+    json data;
+};
+
 class ReactImgui : public ImPlotView {
     private:
         int m_mapGeneratorJobCounter = 0;
         std::unordered_map<int, std::unique_ptr<MapGenerator>> m_mapGeneratorJobs;
 
+        rpp::subjects::replay_subject<ElementOpDef> m_widgetOpSubject;
+        std::mutex m_widgetOpSubjectsMutex;
```

Comment:
just minor tip: if only purpose of this mutex to forbid async on_next calls, you can just use [serialized_replay_subject](https://victimsnino.github.io/ReactivePlusPlus/v2/docs/html/classrpp_1_1subjects_1_1serialized__replay__subject.html#details) instead =)

Reply:
Hi @victimsnino I thought I'd make you aware of this: https://youtrack.jetbrains.com/issue/RSCPP-36040/Resolve-errors-in-ReactivePlusPlus-library

Reply:
Recently I started using CLion and saw that issue, looks like it will get resolved.

Reply:
Hey @andreamancuso , thank you for letting me know. Looks pretty strange.. I think, heavy template magic is reason of this =C

Reply:
<img width="987" alt="Screenshot 2024-08-02 at 16 41 02" src="https://github.com/user-attachments/assets/43d87848-0c70-4b63-8e0b-eca7250abeae">
clangd server could handle it easily.. as far as i remeber, clion has its own LSP to parse c++. Looks like it can be a reason